File: packages/types/src/hermes-reasoning.test.ts
import { describe, it, expect } from 'vitest'
import { splitHermesReasoning, stripAnsi } from './hermes-reasoning.js'

const HEADER =
  '┌─ Reasoning ──────────────────────────────────────────────────────────────────────────────────────┐'
const FOOTER =
  '└──────────────────────────────────────────────────────────────────────────────────────────────────┘'

describe('splitHermesReasoning', () => {
  it('returns empty input unchanged', () => {
    expect(splitHermesReasoning('')).toEqual({ reasoning: '', text: '' })
  })

  it('leaves a normal reply alone', () => {
    expect(splitHermesReasoning('The parser is in src/parse.ts.')).toEqual({
      reasoning: '',
      text: 'The parser is in src/parse.ts.',
    })
  })

  it('strips the TUI box and keeps the reply', () => {
    const raw = [HEADER, '│ The user wants the leak fixed.', '│ Check the den hook.', FOOTER, '', 'Fixed.'].join(
      '\n',
    )
    expect(splitHermesReasoning(raw)).toEqual({
      reasoning: 'The user wants the leak fixed.\nCheck the den hook.',
      text: 'Fixed.',
    })
  })

  it('quiet-mode header with no │ body stops at the blank line', () => {
    const raw = [HEADER, 'The user asked about X.', '', 'Here is the answer.'].join('\n')
    expect(splitHermesReasoning(raw)).toEqual({
      reasoning: 'The user asked about X.',
      text: 'Here is the answer.',
    })
  })

  it('a box-only payload is all reasoning', () => {
    const raw = [HEADER, '│ only thinking', FOOTER].join('\n')
    expect(splitHermesReasoning(raw)).toEqual({ reasoning: 'only thinking', text: '' })
  })

  it('still matches when the TUI wraps the header in ANSI', () => {
    const raw = [`\u001b[90m${HEADER}\u001b[0m`, '│ think', FOOTER, 'hi'].join('\n')
    expect(splitHermesReasoning(raw)).toEqual({ reasoning: 'think', text: 'hi' })
    expect(stripAnsi(`\u001b[31mred\u001b[0m`)).toBe('red')
  })
})

File: apps/rivet-bots-android/app/src/test/java/dev/rivetos/bots/HermesReasoningTest.kt
package dev.rivetos.bots

import dev.rivetos.bots.data.splitHermesReasoning
import dev.rivetos.bots.data.visibleAssistantText
import org.junit.Assert.assertEquals
import org.junit.Test

class HermesReasoningTest {
    private val header =
        "┌─ Reasoning ──────────────────────────────────────────────────────────────────────────────────────┐"
    private val footer =
        "└──────────────────────────────────────────────────────────────────────────────────────────────────┘"

    @Test fun `normal reply is untouched`() {
        val s = splitHermesReasoning("The parser is in src/parse.ts.")
        assertEquals("", s.reasoning)
        assertEquals("The parser is in src/parse.ts.", s.text)
    }

    @Test fun `tui box is stripped from the reply`() {
        val raw = listOf(header, "│ The user wants the leak fixed.", footer, "", "Fixed.").joinToString("\n")
        val s = splitHermesReasoning(raw)
        assertEquals("The user wants the leak fixed.", s.reasoning)
        assertEquals("Fixed.", s.text)
        assertEquals("Fixed.", visibleAssistantText(raw))
    }

    @Test fun `box-only payload has no visible text`() {
        val raw = listOf(header, "│ only thinking", footer).joinToString("\n")
        assertEquals("", visibleAssistantText(raw))
    }
}
